File: app/ios/RunninWatch Watch App/Screens/RunCompletedScreen.swift
import SwiftUI
import WatchConnectivity

/// Mostrada após corrida concluída (status: .completed).
/// Espelha a ReportPage do iPhone — totais + splits — adaptado pro Watch.
///
/// Layout:
///   - Header "CORRIDA CONCLUÍDA" em accentColor
///   - Stats principais: DIST (secondary, big), TEMPO (white, big), PACE MED
///     (primary, big)
///   - Secondary: BPM médio, calorias, elevação
///   - Lista de splits (scrollable)
///   - Botão "OK" no rodapé — manda `acknowledgeComplete` pro iPhone que
///     transiciona pra idle (TypeSelectorScreen)
struct RunCompletedScreen: View {
    @EnvironmentObject var state: WatchRunState

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 6) {
                HStack {
                    RunninLogo()
                    Spacer()
                }
                .padding(.top, 2)

                Text("CORRIDA CONCLUÍDA")
                    .font(state.scaledFont(size: 9, weight: .heavy))
                    .tracking(1.5)
                    .foregroundStyle(state.accentColor)
                    .padding(.top, 2)

                // Stats principais — mesma hierarquia cromática da ActiveRunScreen
                statBlock(label: "DIST", value: state.formattedDistance,
                          unit: "km", size: 26, color: state.secondaryColor)
                statBlock(label: "TEMPO", value: state.formattedElapsed,
                          unit: nil, size: 22, color: .white)
                statBlock(label: "PACE MED", value: state.formattedPace,
                          unit: "/km", size: 22, color: state.accentColor)

                // Stats secundários compactos numa row 3-col
                HStack(spacing: 6) {
                    miniStat(label: "BPM",
                             value: state.bpm > 0 ? "\(state.bpm)" : "—",
                             color: state.secondaryColor)
                    miniStat(label: "CAL",
                             value: "\(Int(state.caloriesKcal))",
                             color: .white)
                    miniStat(label: "ELEV",
                             value: "+\(Int(state.elevationM))m",
                             color: .white.opacity(0.8))
                }

                // Splits (se houver)
                if !state.splits.isEmpty {
                    Text("SPLITS")
                        .font(state.scaledFont(size: 8, weight: .medium))
                        .tracking(1.5)
                        .foregroundStyle(.white.opacity(0.5))
                        .padding(.top, 4)
                    ForEach(state.splits) { split in
                        splitRow(split)
                    }
                }

                // Botão OK
                Button(action: acknowledge) {
                    Text("OK")
                        .font(state.scaledFont(size: 13, weight: .heavy))
                        .tracking(1.2)
                        .foregroundStyle(.black)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 9)
                        .background(state.accentColor)
                        .clipShape(RoundedRectangle(cornerRadius: 6))
                }
                .buttonStyle(.plain)
                .padding(.top, 8)
            }
            .padding(.horizontal, 6)
            .padding(.bottom, 6)
        }
    }

    @ViewBuilder
    private func statBlock(label: String, value: String, unit: String?,
                           size: CGFloat, color: Color) -> some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(label)
                .font(state.scaledFont(size: 7, weight: .medium))
                .tracking(0.8)
                .foregroundStyle(.white.opacity(0.5))
            HStack(alignment: .firstTextBaseline, spacing: 3) {
                Text(value)
                    .font(state.scaledFont(size: size, weight: .bold))
                    .foregroundStyle(color)
                    .minimumScaleFactor(0.5)
                    .lineLimit(1)
                if let unit = unit {
                    Text(unit)
                        .font(state.scaledFont(size: 9, weight: .medium))
                        .foregroundStyle(.white.opacity(0.5))
                }
            }
        }
    }

    @ViewBuilder
    private func miniStat(label: String, value: String, color: Color) -> some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(label)
                .font(state.scaledFont(size: 6, weight: .medium))
                .tracking(0.8)
                .foregroundStyle(.white.opacity(0.45))
            Text(value)
                .font(state.scaledFont(size: 12, weight: .bold))
                .foregroundStyle(color)
                .minimumScaleFactor(0.5)
                .lineLimit(1)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    @ViewBuilder
    private func splitRow(_ split: WatchSplit) -> some View {
        HStack(spacing: 5) {
            Text("KM\(split.km)")
                .font(state.scaledFont(size: 10, weight: .heavy))
                .foregroundStyle(state.accentColor)
                .frame(width: 28, alignment: .leading)
            VStack(alignment: .leading, spacing: 0) {
                Text(split.pace)
                    .font(state.scaledFont(size: 11, weight: .bold))
                    .foregroundStyle(.white)
                Text(split.formattedDuration)
                    .font(state.scaledFont(size: 7, weight: .medium))
                    .foregroundStyle(.white.opacity(0.5))
            }
            Spacer(minLength: 0)
            if split.bpm > 0 {
                Text("\(split.bpm)")
                    .font(state.scaledFont(size: 10, weight: .bold))
                    .foregroundStyle(state.secondaryColor)
            }
        }
        .padding(.horizontal, 5)
        .padding(.vertical, 4)
        .background(Color.white.opacity(0.05))
        .clipShape(RoundedRectangle(cornerRadius: 3))
    }

    /// User tocou OK — manda `acknowledgeComplete` pro iPhone, que vai
    /// transicionar de volta pra idle (push status=idle). Watch reseta o
    /// localStep pra .selectingType ao receber.
    private func acknowledge() {
        guard WCSession.isSupported() else { return }
        let session = WCSession.default
        guard session.activationState == .activated else { return }
        let msg: [String: Any] = ["action": "acknowledgeComplete"]
        if session.isReachable {
            session.sendMessage(msg, replyHandler: nil, errorHandler: { _ in
                session.transferUserInfo(msg)
            })
        } else {
            session.transferUserInfo(msg)
        }
        // Otimista: força localStep idle imediatamente — UI volta pro
        // TypeSelector mesmo se o push de volta do iPhone demorar.
        state.status = .idle
        state.localStep = .selectingType
        state.splits = []
    }
}

File: app/android/wear/src/main/kotlin/com/s6lab/runnin/wear/screens/RunCompletedScreen.kt
package com.s6lab.runnin.wear.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.wear.compose.material.Text
import com.s6lab.runnin.wear.bridge.WearableMessenger
import com.s6lab.runnin.wear.components.RunninLogo
import com.s6lab.runnin.wear.models.LocalStep
import com.s6lab.runnin.wear.models.WatchRunState
import com.s6lab.runnin.wear.models.WatchSplit
import com.s6lab.runnin.wear.models.WatchStatus
import com.s6lab.runnin.wear.theme.scaledFont

/**
 * Mostrada após corrida concluída (status: completed). Espelha a ReportPage
 * do phone, adaptada pro Watch.
 *
 * Paralelo do `RunCompletedScreen` do Apple Watch (Screens/RunCompletedScreen.swift).
 */
@Composable
fun RunCompletedScreen() {
    val state = WatchRunState.shared
    val accent = state.accentColor
    val scroll = rememberScrollState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(scroll)
            .padding(horizontal = 6.dp, vertical = 6.dp),
        verticalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Row(Modifier.fillMaxWidth().padding(top = 2.dp)) {
            RunninLogo()
            Spacer(Modifier.weight(1f))
        }
        Text(
            text = "CORRIDA CONCLUÍDA",
            color = accent,
            style = scaledFont(9f, FontWeight.Black),
            modifier = Modifier.padding(top = 2.dp),
        )

        StatBlock(label = "DIST", value = state.formattedDistance, unit = "km",
            size = 26f, color = state.secondaryColor)
        StatBlock(label = "TEMPO", value = state.formattedElapsed, unit = null,
            size = 22f, color = Color.White)
        StatBlock(label = "PACE MED", value = state.formattedPace, unit = "/km",
            size = 22f, color = accent)

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(6.dp),
        ) {
            MiniStat(label = "BPM",
                value = if (state.bpm > 0) "${state.bpm}" else "—",
                color = state.secondaryColor,
                modifier = Modifier.weight(1f))
            MiniStat(label = "CAL", value = "${state.caloriesKcal.toInt()}",
                color = Color.White, modifier = Modifier.weight(1f))
            MiniStat(label = "ELEV", value = "+${state.elevationM.toInt()}m",
                color = Color.White.copy(alpha = 0.8f), modifier = Modifier.weight(1f))
        }

        if (state.splits.isNotEmpty()) {
            Text(
                text = "SPLITS",
                color = Color.White.copy(alpha = 0.5f),
                style = scaledFont(8f, FontWeight.Medium),
                modifier = Modifier.padding(top = 4.dp),
            )
            state.splits.forEach { CompletedSplitRow(it) }
        }

        // Botão OK
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(6.dp))
                .background(accent)
                .clickable {
                    WearableMessenger.sendCommand("acknowledgeComplete")
                    // Otimista: força localStep idle imediatamente.
                    state.localStep = LocalStep.SelectingType
                }
                .padding(vertical = 9.dp),
            contentAlignment = Alignment.Center,
        ) {
            Text("OK", color = Color.Black, style = scaledFont(13f, FontWeight.Black))
        }
    }
}

@Composable
private fun StatBlock(label: String, value: String, unit: String?, size: Float, color: Color) {
    Column {
        Text(label, color = Color.White.copy(alpha = 0.5f),
            style = scaledFont(7f, FontWeight.Medium))
        Row(verticalAlignment = Alignment.Bottom, horizontalArrangement = Arrangement.spacedBy(3.dp)) {
            Text(value, color = color, style = scaledFont(size, FontWeight.Bold), maxLines = 1)
            unit?.let {
                Text(it, color = Color.White.copy(alpha = 0.5f),
                    style = scaledFont(9f, FontWeight.Medium))
            }
        }
    }
}

@Composable
private fun MiniStat(label: String, value: String, color: Color, modifier: Modifier = Modifier) {
    Column(modifier = modifier) {
        Text(label, color = Color.White.copy(alpha = 0.45f),
            style = scaledFont(6f, FontWeight.Medium))
        Text(value, color = color, style = scaledFont(12f, FontWeight.Bold), maxLines = 1)
    }
}

@Composable
private fun CompletedSplitRow(split: WatchSplit) {
    val state = WatchRunState.shared
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(3.dp))
            .background(Color.White.copy(alpha = 0.05f))
            .padding(horizontal = 5.dp, vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(5.dp),
    ) {
        Text("KM${split.km}", color = state.accentColor,
            style = scaledFont(10f, FontWeight.Black),
            modifier = Modifier.width(28.dp))
        Column(Modifier.weight(1f)) {
            Text(split.pace, color = Color.White, style = scaledFont(11f, FontWeight.Bold))
            Text(split.formattedDuration, color = Color.White.copy(alpha = 0.5f),
                style = scaledFont(7f, FontWeight.Medium))
        }
        if (split.bpm > 0) {
            Text("${split.bpm}", color = state.secondaryColor,
                style = scaledFont(10f, FontWeight.Bold))
        }
    }
}
